omnibus/files/private-chef-cookbooks/private-chef/libraries/private_chef.rb

```diff
@@ -435,60 +426,74 @@ def gen_frontend_default
     end
 
     def gen_api_fqdn_default
-      PrivateChef["lb"]["api_fqdn"] ||= PrivateChef['api_fqdn']
-      PrivateChef["lb"]["web_ui_fqdn"] ||= PrivateChef['api_fqdn']
-      PrivateChef["nginx"]["server_name"] ||= PrivateChef['api_fqdn']
+      PrivateChef["lb"]["api_fqdn"] ||= PrivateChef["api_fqdn"]
+      PrivateChef["lb"]["web_ui_fqdn"] ||= PrivateChef["api_fqdn"]
+      PrivateChef["nginx"]["server_name"] ||= PrivateChef["api_fqdn"]
       PrivateChef["nginx"]["url"] ||= "https://#{PrivateChef['api_fqdn']}"
     end
 
     def gen_secrets_default(node_name)
-      consume_existing_secrets
-      # Transition from erchef's sql_user/password etc living under 'postgresql' in older versions,
-      # to 'opscode_erchef' in newer versions.
-      if PrivateChef['postgresql'].has_key? 'sql_password'
-        PrivateChef['opscode_erchef']['sql_password'] ||= PrivateChef['postgresql']['sql_password']
-        PrivateChef['postgresql'].delete 'sql_password'
-        PrivateChef['postgresql'].delete 'sql_user'
+      secrets_json = "/etc/opscode/private-chef-secrets.json"
+      credentials =
+        if File.exist?(secrets_json)
+          Veil::CredentialCollection::ChefSecretsFile.from_file(secrets_json)
+        elsif PrivateChef["topology"] == "ha" && !PrivateChef["servers"][node_name]["bootstrap"]
+          Chef::Log.fatal("In an H/A topology the secrets must be created on the bootstrap node. "\
+                          "Please copy the contents of /etc/opscode/ from your bootstrap Server " \
+                          "to complete the setup")
+          exit(44)
+        else
+          Veil::CredentialCollection::ChefSecretsFile.new(path: secrets_json)
```

Comment:
This gets a little odd with the external-backends + standalone-as-frontends topo. We have not addressed the upgrade scenario for that case yet, but in the initial setup the first standalone node you bring up behaves as bootstrapping, while subsequent nodes will be expected to have secrets copied over. 

Worth mentioning here because the topology looks like 'standalone' in that case.